ui/src/components/modals/ProcessModal.tsx

```diff
@@ -38,15 +39,17 @@ const ProcessModal: React.FC<Props> = ({position, closeModal, handleAddNode}) =>
 
   const handleSubmit = async (e) => {
     e.preventDefault()
-    const process: Process =  new Process({name, finished, note, classifiedAs, inScopeOf, basedOn} as ProcessShape);
-    const planId = getDataStore().getRoot()['planId'];
-    const path: string = `root.plan.${planId}.process.${process.id}`;
-    const input: ThingInput = {
-      path,
-      data: JSON.stringify(process)
-    }
-    await getZomeApi().put_thing(input);
-    handleAddNode(input);
+
+    const store = getDataStore();
+    const plannedWithin = store.getCursor('root.planId');
+    const process: Process = new Process(
+      rejectEmptyFields<ProcessShape>(
+        {name, plannedWithin, finished, note, classifiedAs, inScopeOf, basedOn}
+      )
+    );
+    await store.set(process);
+    handleAddNode(process);
```

Comment:
This is going to be the way we should do this, but I'm thinking of moving the reject empty fields into the models' constructors themselves. Then we won't have to think about it.

Reply:
Cool, will do.